cli/result.go

```diff
@@ -29,16 +29,33 @@ func (cli *CLI) NewResultsCmd() *cobra.Command {
 
 			pemBlock, _ := pem.Decode(privKeyFile)
 
-			privKey, err := x509.ParsePKCS1PrivateKey(pemBlock.Bytes)
-			if err != nil {
-				log.Fatalf("Error parsing private key: %v", err)
-			}
+			var result []byte
 
-			result, err := cli.agentSDK.Result(cmd.Context(), privKey)
-			if err != nil {
-				log.Fatalf("Error retrieving computation result: %v", err)
+			switch pemBlock.Type {
+			case rsaKeyType:
+				privKey, err := x509.ParsePKCS8PrivateKey(pemBlock.Bytes)
+				if err != nil {
+					privKey, err = x509.ParsePKCS1PrivateKey(pemBlock.Bytes)
+					if err != nil {
+						log.Fatalf("Error parsing private key: %v", err)
+					}
+				}
+				result, err = cli.agentSDK.Result(cmd.Context(), privKey)
+				if err != nil {
+					log.Fatalf("Error retrieving computation result: %v", err)
+				}
+			case ecdsaKeyType:
```

Comment:
what about ed2

Reply:
Ed2 shares the same type as rsa, so both are handled in the rsaKeyType case.